pkg/controller/perconaservermongodb/psmdb_controller.go

```diff
@@ -430,6 +430,11 @@ func (r *ReconcilePerconaServerMongoDB) Reconcile(request reconcile.Request) (re
 		}
 	}
 
+	err = r.doInCaseOfRestore(cr)
+	if err != nil {
+		return reconcile.Result{}, errors.Wrap(err, "on restore")
+	}
+
```

Comment:
I think we should delete stop balancer and delete mongos from `runRestore` function.
Why we run `doInCaseOfRestore` here?